src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +29,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
+        return Objects.equals(width, rectangle.width)
+                && Objects.equals(length, rectangle.length)
+                && Objects.equals(color, rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        if (width == null || length == null || color == null) {
+            return 0;
+        }
```

Comment:
if any of your fields is null, you ignore all the others and return null
this isn't right

Reply:
but it`s required to return int not null

Reply:
sorry, I meant 
`if any of your fields is null, you ignore all the others and return 0
this isn't right`

the logic itself is not correct
you should decide whether to change `result` after checking each field, and not them all together